frontend/test/resolution/testReturnTypes.cpp

```diff
@@ -1094,6 +1094,172 @@ static void testSelectParams() {
   }
 }
 
+// Assumes loop body will return param string "asdf"
+static void paramLoopTestHelper(Context* context, const char* loopBody,
+                                bool returnFromLoop, bool useEmptyLoop = false) {
+  context->advanceToNextRevision(false);
+  ErrorGuard guard(context);
+
+  // Construct test program
+  std::ostringstream oss;
+  oss << ops;
+  oss << "proc foo() param {\n";
+  oss << "  for param i in " << (useEmptyLoop ? "1..0" : "0..2") << " {\n";
```

Comment:
Using this bool here makes the test nonlocal (I can't easily figure out the bounds from the test invocation, and the invocation contains logic that matches on the bounds). Could you just accept the range `const char*` as an argument to this test? That way, when I'm looking at the loop body in `testParamLoop`, I can see that `if i == 1` will be run with `0..2`.

Reply:
Yeah, I'll change it. My reasoning was that the range doesn't matter to the user of the helper, except whether it's empty or non-empty, but that doesn't hold up for some of the test cases as you point out.